pkg/rules/service_test.go

```diff
@@ -0,0 +1,70 @@
+package rules
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+
+	commonmonitoring "github.com/giantswarm/observability-operator/pkg/common/monitoring"
+)
+
+func TestGenerateAlloyConfig(t *testing.T) {
+	tests := []struct {
+		name       string
+		tenants    []string
+		goldenPath string
+	}{
+		{
+			name:       "TwoTenants",
+			tenants:    []string{"tenant1", "tenant2"},
+			goldenPath: filepath.Join("testdata", "alloy_config_multitenants.river"),
+		},
+		{
+			name:       "SingleTenant",
+			tenants:    []string{"tenant1"},
+			goldenPath: filepath.Join("testdata", "alloy_config_singletenant.river"),
+		},
+		{
+			name:       "DefaultTenantRendersLegacyConfig",
+			tenants:    []string{commonmonitoring.DefaultWriteTenant},
+			goldenPath: filepath.Join("testdata", "alloy_config_defaulttenant.river"),
+		},
```

Comment:
What happens when there's no tenant ? Is the config valid ? That's a weird corner case though.

Reply:
ok fine